pkg/cloudprovider/vsphere/cloud_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	ccfg "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/config"
+	vcfg "k8s.io/cloud-provider-vsphere/pkg/common/config"
+)
+
+func Test_validateDualStack(t *testing.T) {
+	testcases := []struct {
+		name          string
+		gateEnabled   bool
+		cfg           *ccfg.CPIConfig
+		expectedError error
+	}{
+		{
+			name:        "config dual-stack, gate enabled",
+			gateEnabled: true,
+			cfg: &ccfg.CPIConfig{
+				Config: vcfg.Config{
+					VirtualCenter: map[string]*vcfg.VirtualCenterConfig{
+						"vcenter.local": {
+							IPFamilyPriority: []string{"v4", "v6"},
```

Comment:
should we consider the testcase with an empty` IPFamilyPriority`?

Reply:
added a testcase for the default config